lib/superagent-mock.js

```diff
@@ -109,15 +109,18 @@ function mock (superagent, config) {
           res: {
             headers: {},
             setEncoding: function (){},
-            on: function (){}
+            on: function (){},
+            body: err.responseBody || {}
           },
           req: {
             method: function (){}
           },
+
           xhr: {
             responseType: '',
+            responseText: err.responseText || '',
```

Comment:
Client-side, `response.body` gets created by parsing `responseText` and `responseHeader` - see [here](https://github.com/visionmedia/superagent/blob/master/lib/client.js#L316) 
Server-side, `response.body` gets just assigned to whatever the body of the response already is - see [here](https://github.com/visionmedia/superagent/blob/master/lib/node/response.js#L38)

I needed to include both for all tests to pass :)